plugins/destination/bigquery/client/read.go

```diff
@@ -94,9 +90,9 @@ func (c *Client) Read(ctx context.Context, table *schema.Table, sourceName strin
 	if err != nil {
 		return fmt.Errorf("failed to read table %s: %w", table.Name, err)
 	}
-	values := c.createResultsArray(table)
-	v := make([]any, len(values))
 	for {
+		values := c.createResultsArray(table)
+		v := make([]interface{}, len(values))
```

Comment:
Why does this need to be inside the for loop - we should only need to initialize these once I think?

Reply:
it was a bug apparently. as it was returning the same object over the channel overriding results everytime it receives something to all results.